nightly-examples/examples/spawny_thing.rs

```diff
@@ -0,0 +1,47 @@
+#![deny(rust_2018_idioms)]
+
+/// This is a example showing how information is scoped.
+///
+/// You can run this example by running the following command in a terminal
+///
+/// ```
+/// cargo +nightly run --example spawny_thing
+/// ```
+use tokio;
+
+use futures::future::join_all;
+use tracing::{debug, info};
+use tracing_attributes::instrument;
+
+#[instrument]
+async fn parent_task(subtasks: usize) {
+    info!("spawning subtasks...");
+    let subtasks = (1..=subtasks)
+        .map(|number| {
+            debug!(message = "creating subtask;", number);
+            subtask(number)
+        })
+        .collect::<Vec<_>>();
+
+    let result = join_all(subtasks).await;
+
+    debug!("all subtasks completed");
+    let sum: usize = result.into_iter().sum();
+    info!(sum);
+}
+
+#[instrument]
+async fn subtask(number: usize) -> usize {
+    info!("polling subtask... {}", number);
```

Comment:
Note that because we `instrument` this function we don't actually need to record the number at all --- it will be inherited from the span.
```suggestion
    info!("polling subtask...");
```